packages/back-end/src/util/migrations.ts

```diff
@@ -691,6 +745,33 @@ export function migrateSnapshot(
       attributionModel: "firstExposure",
       variations,
     };
+  } else {
+    // Add new settings field in case it is missing
+    if (snapshot.settings.defaultMetricPriorSettings === undefined) {
+      snapshot.settings.defaultMetricPriorSettings = defaultMetricPriorSettings;
+    }
+
+    // migrate metric for snapshot to have new fields as old snapshots
+    // may not have prior settings
+    snapshot.settings.metricSettings = snapshot.settings.metricSettings.map(
+      (m) => {
+        if (m.computedSettings) {
+          m.computedSettings = {
+            ...m.computedSettings,
+            properPrior:
+              m.computedSettings.properPrior ??
+              defaultMetricPriorSettings.proper,
+            properPriorMean:
+              m.computedSettings.properPriorMean ??
+              defaultMetricPriorSettings.mean,
+            properPriorStdDev:
+              m.computedSettings.properPriorStdDev ??
+              defaultMetricPriorSettings.stddev,
```

Comment:
Minor code style. You can control default and override behavior based on where you put the spread operator.

```ts
{
  a: "default",
  ...obj,
  b: "override"
}
```

obj.a will take precedence over the default, but obj.b will be ignored in favor of the override.